### DISCUSSION TITLE
json_pointer not functioning as expected from docs

### DISCUSSION THREAD

**Discussion by clarkholdham** [nlohmann/json#3511] [Q&A] [answered]
`json::json_pointer` access not working as expected from [here](https://json.nlohmann.me/features/json_pointer/#value-access) with root "/". I've tried with the string literal as well as standard `json::json_pointer` constructors as well. 

Code:
```
#include <nlohmann/json.hpp>

#include <iostream>

int main() {
    nlohmann::json data = {{"data", {}}};
    nlohmann::json::json_pointer p = "/"_json_pointer;
    std::cout << "No pointer: " << data.dump() << std::endl;
    try {
        std::cout << "Checked access pointer: " << data.at(p).dump()
                  << std::endl;
    } catch (std::exception& e) {
        std::cout << "Json error: " << e.what() << std::endl;
    }
    try {
        std::cout << "Unchecked access pointer: " << data[p].dump()
                  << std::endl;
    } catch (std::exception& e) {
        std::cout << "Json error: " << e.what() << std::endl;
    }
}
```

Actual:
```
No pointer: {"data":null}
Checked access pointer: Json error: [json.exception.out_of_range.403] key '' not found
Unchecked access pointer: null
```
Expected:
```
No pointer: {"data":null}
Checked access pointer: {"data":null}
Unchecked access pointer: {"data":null}
```

**Comment by falbrechtskirchinger** [comment, marked as answer]
The JSON pointer `"/"` refers to the object member with the key `""`. From the [RFC](https://datatracker.ietf.org/doc/html/rfc6901#section-5):

> For example, given the JSON document
> ```json
>    {
>       "foo": ["bar", "baz"],
>       "": 0,
>       "a/b": 1,
>       "c%d": 2,
>       "e^f": 3,
>       "g|h": 4,
>       "i\\j": 5,
>       "k\"l": 6,
>       " ": 7,
>       "m~n": 8
>    }
> ```
>    The following JSON strings evaluate to the accompanying values:
> 
> ```json
>     ""           // the whole document
>     "/foo"       ["bar", "baz"]
>     "/foo/0"     "bar"
>     "/"          0
>     "/a~1b"      1
>     "/c%d"       2
>     "/e^f"       3
>     "/g|h"       4
>     "/i\\j"      5
>     "/k\"l"      6
>     "/ "         7
>     "/m~0n"      8
> ```

Edit: The documentation you referenced is in fact wrong.
